misc/debug-adapter/modules/debug-adapter-runtime/src/main/java/org/ballerinalang/debugadapter/runtime/DebuggerRuntime.java

```diff
@@ -0,0 +1,157 @@
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.concurrent.CountDownLatch;
+import java.util.function.Function;
+
+/**
+ * This class contains the set of runtime helper util methods to support debugger expression evaluation.
+ * <p>
+ * These utils methods must be class-loaded into the program JVM to evaluate
+ * <ul>
+ *  <li> function
+ *  <li> object method
+ *  <li> remote call action
+ *  <li> wait action
+ *  </ul>
+ * invocations using the debugger expression evaluation engine.
+ *
+ * @since 2.0.0
+ */
+@SuppressWarnings("unused")
+public class DebuggerRuntime {
+
+    private static final String EVALUATOR_STRAND_NAME = "evaluator-strand";
+
+    /**
+     * Invokes Ballerina object methods in blocking manner.
+     *
+     * @param bObject    ballerina object instance
+     * @param methodName name of the object method to be invoked
+     * @param args       object method arguments
+     * @return return values
+     */
+    public static Object invokeObjectMethod(BObject bObject, String methodName, Object... args) {
+        try {
+            Scheduler scheduler = new Scheduler(1, false);
+            CountDownLatch latch = new CountDownLatch(1);
+            final Object[] finalResult = new Object[1];
+            final Object[] paramValues = args[0] instanceof Strand ? Arrays.copyOfRange(args, 1, args.length) : args;
+
+            Function<?, ?> func = o -> bObject.call((Strand) (((Object[]) o)[0]), methodName, paramValues);
+            Object resultFuture = scheduler.schedule(new Object[1], func, null, new Callback() {
+                @Override
+                public void notifySuccess(Object result) {
+                    latch.countDown();
+                    finalResult[0] = result;
+                }
+
+                @Override
+                public void notifyFailure(BError error) {
+                    latch.countDown();
+                    finalResult[0] = error;
+                }
+            }, EVALUATOR_STRAND_NAME, null).result;
+
+            scheduler.start();
+            latch.await();
+            return finalResult[0];
+        } catch (Exception e) {
+            throw new BallerinaException("invocation failed: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Invoke Ballerina functions in blocking manner.
+     *
+     * @param classLoader normal classLoader
+     * @param className   which the function resides/ or file name
+     * @param methodName  to be invokable unit
+     * @param paramValues to be passed to invokable unit
+     * @return return values
+     */
+    public static Object invokeFunction(ClassLoader classLoader, String className, String methodName,
+                                        Object... paramValues) {
+        try {
+            Scheduler scheduler = new Scheduler(1, false);
+            Class<?> clazz = classLoader.loadClass(className);
```

Comment:
Shall we add a meaningful name instead of `clazz`?

Reply:
@praveennadarajah `clazz` is commonly used https://stackoverflow.com/questions/2529974/why-do-java-programmers-like-to-name-a-variable-clazz/2530174

Reply:
Ack